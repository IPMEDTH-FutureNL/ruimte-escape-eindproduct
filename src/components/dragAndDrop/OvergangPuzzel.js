import React from 'react'
import { useNavigate } from "react-router-dom";
import IntroSound from '../../sound/Intro.mp3';

const OvergangPuzzel = () => {

    const navigate = useNavigate();
    const audio = new Audio(IntroSound);
    
    const goToPuzzle = () => {
        audio.pause()
        navigate("/puzzelEen");
    }

    return(
        <section className="container__intro-puzzelEen">
            <div className="intro__container-text" id="text">
                <p className="intro__container-paragraph">
                We willen graag vertrekken naar Mars. Maar de planeten staan verkeerd op de kaart.
                </p>
            </div>
            <button className="button primary center" onClick={() => goToPuzzle()}>Toon kaart!</button>
        </section>
    )
}

export default OvergangPuzzel;