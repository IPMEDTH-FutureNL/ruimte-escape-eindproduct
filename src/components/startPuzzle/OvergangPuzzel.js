import React from 'react'
import { useNavigate } from "react-router-dom";

const OvergangPuzzel = () => {

    const navigate = useNavigate();
    
    const goToPuzzle = () => {
        navigate("/puzzelEen");
    }

    const video = () => {
        let videoPlaceholder = document.getElementById("video");
        let textPlaceholder = document.getElementById("text");
        videoPlaceholder.style.display = "block";
        textPlaceholder.style.display = "none";
    }

    return(
        <section className="container__intro-puzzelEen">
            <div className="intro__container-text" id="text">
                <p className="intro__container-paragraph">
                    We willen gaan opstijgen, maar
                    de planeten op de kaart
                    staan niet in de goede volgorde.
                    Aan jou de taak om ze juist op de kaart
                    te zetten!
                </p>
                <button className="button primary" onClick={() => video()}>Toon kaart!</button>
            </div>
            <div className="intro__container-video" id="video">
                <div className="intro__container-videoPlaceholder">
                    <p className="placeHolder">Placeholder voor video eventueel</p>
                </div>
                <button className="button primary" onClick={() => goToPuzzle()}>Naar puzzel!</button>
            </div>
        </section>
    )
}

export default OvergangPuzzel;