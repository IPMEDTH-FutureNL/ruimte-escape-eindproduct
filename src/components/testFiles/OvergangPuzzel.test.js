import renderer from 'react-test-renderer';
import OvergangPuzzel from "../dragAndDrop/OvergangPuzzel";
import {BrowserRouter as Router} from 'react-router-dom';

it('Test if Overgang Puzzel component is rendered', () => {
    const component = renderer.create(
        <Router>
            <OvergangPuzzel></OvergangPuzzel>,
        </Router>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})