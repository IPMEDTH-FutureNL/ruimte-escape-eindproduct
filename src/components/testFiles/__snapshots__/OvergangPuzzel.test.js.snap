// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test if Overgang Puzzel component is rendered 1`] = `
Array [
  <section
    className="container__intro-puzzelEen"
  >
    <div
      className="intro__container-text"
      id="text"
    >
      <p
        className="intro__container-paragraph"
      >
        We willen gaan opstijgen, maar de planeten op de kaart staan niet in de goede volgorde. Aan jou de taak om ze juist op de kaart te zetten!
      </p>
      <button
        className="button primary"
        onClick={[Function]}
      >
        Toon kaart!
      </button>
    </div>
  </section>,
  ",",
]
`;
